### PR TITLE
Add e2e runner for conformance

### PR DESCRIPTION
**Pull request by wackxu**
**What type of PR is this?**

/kind feature
**What this PR does / why we need it**:

Add e2e runner for that we can run e2e easily on user cluster

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: build/conformance/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+COPY --from=builder /usr/local/bin/e2e-runner /usr/local/bin/e2e-runner
+
+COPY --from=builder /go/src/github.com/kubeedge/kubeedge/tests/e2e/e2e.test /usr/local/bin/e2e.test
+
+COPY --from=builder  /go/src/github.com/kubeedge/kubeedge/build/conformance/kubernetes/edge_skip_case.yaml /testdata/edge_skip_case.yaml
+
+RUN mkdir -p /tmp/results
+
+ENTRYPOINT ["e2e-runner"]
```
Comment (gy95) [inline build/conformance/Dockerfile:45]: add a deployment sample? to illustrate e2e-runner cmd flags

File: build/conformance/e2e-runner/run.go [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+)
+
+const (
+	dryRunEnvKey        = "E2E_DRYRUN"
+	skipEnvKey          = "E2E_SKIP"
+	ginkgoEnvKey        = "GINKGO_BIN"
+	testBinEnvKey       = "TEST_BIN"
+	resultsDirEnvKey    = "RESULTS_DIR"
+	kubeConfigEnvKey    = "KUBECONFIG"
+	logFileName         = "e2e.log"
```
Comment (gy95) [inline build/conformance/e2e-runner/run.go:54]: need this saved on the machine, but not only in container?

Comment (vincentgoat) [reply build/conformance/e2e-runner/run.go:58]: yes

File: build/conformance/Dockerfile (hunk 1 above)
Comment (vincentgoat) [reply build/conformance/Dockerfile:45]: done

**Pull request merged**